text/0018-simplify-installer.md

```diff
@@ -0,0 +1,152 @@
+
+```
+$ kubectl -n keptn get service bridge api-gateway-nginx
+NAME                TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)          AGE
+bridge              ClusterIP   10.0.83.209   <none>        8080:32122/TCP   85d
+api-gateway-nginx   ClusterIP   10.0.94.251   <none>        80/TCP           41d
+```
+In this example neither bridge nor API are reachable from the outside. To reach it, one would have to 
+
+Here is another example if using `keptn install --keptn-api-service-type=LoadBalancer --keptn-bridge-service-type=ClusterIP`
+```
+$ kubectl -n keptn get service bridge api-gateway-nginx
+NAME                TYPE           CLUSTER-IP    EXTERNAL-IP    PORT(S)          AGE
+bridge              ClusterIP      10.0.83.209   <none>         8080:32122/TCP   85d
+api-gateway-nginx   LoadBalancer   10.0.94.251   11.22.33.44    80/TCP           41d
+```
+In this example, the API is reachable via `http://11.22.33.44:80`.
+
+
+Depending on the selected type, various restrictions might apply, e.g.:
+
+* `LoadBalancer` is not always available and the services external IP stays in status `<Pending>`.
+* `NodePort` might expose the service on a high port number (e.g., 31233) that might need to be configured using firewall rules
+* `ClusteriP` only exposes the service within the cluster, but not to the outside. This setting is desireable when applying an ingress-manifest manually afterwards.
+
+
+## Internal details
+
+* CLI: Remove the install option `--ingress-install-option` and its related code
+* CLI: Remove the install option `--gateway` and its related code
+* CLI: Introduce new options in the CLI for Keptn Install `keptn-api-service-type` and `keptn-bridge-service-type` which default to `ClusterIP` 
+* CLI: After installation, try to access the API via all possible IP addresses that the installer determined. If nothing works, show the user a warning and a link to the troubleshooting guide in our docs.
+* CLI: Installation should be considered successful if all deployments/pods have been started. Being able to reach Keptn API is not a strong requirement, and should only print a warning.
+* Installer: Remove any manifests and any scripts that make use of Istio or nginx-ingress
+* Installer: use a placeholder for the service-type of `api-gateway-nginx` and `bridge`, which is then applied based on the settings `--keptn-api-service-type` and `--keptn-bridge-service-type`
+* Installer: Determine all possible IP addresses (and ports) that the user can use to access Keptn API (internal IP, external IP, ...)
+* Installer: Do not set `KEPTN_DOMAIN`
+* Installer: Do not generate SSL certificates
+
+* CLI/API: Change `keptn configure bridge` no longer provide the flag `action=[expose | lockdown]`, but a flag to configure the service-type [user and password should stay as they are, they are still required]
```

Comment:
> I would propose that neither the CLI nor the API can change the service type because this should be handled by an administrator.

I agree with that proposal. We should document the command, and remove the funcionality from our API.